LongWarOfTheChosen/Src/LW_Overhaul/Classes/X2Ability_LW_AssaultAbilitySet.uc

```diff
@@ -635,3 +639,154 @@ static function X2AbilityTemplate CreateChainLightningAbility()
+
+
+
+	Cooldown = new class'X2AbilityCooldown';
+	Cooldown.iNumTurns = default.CHAIN_LIGHTNING_COOLDOWN;
+	Template.AbilityCooldown = Cooldown;
+
+	// Action Point
+	ActionPointCost = new class'X2AbilityCost_ActionPoints';
+	ActionPointCost.iNumPoints = default.CHAIN_LIGHTNING_MIN_ACTION_REQ;
+	ActionPointCost.DoNotConsumeAllSoldierAbilities.AddItem('Unlimitedpower_LW');
+	ActionPointCost.bConsumeAllPoints = true;
+	Template.AbilityCosts.AddItem(ActionPointCost);	
+
+	//Stun Effect
+	StunnedEffect = CreateArcthrowerStunnedStatusEffect(100);
+	Template.AddTargetEffect(StunnedEffect);
+	Template.AddMultiTargetEffect(StunnedEffect);
+
+	//CleanUpEffect = CreateStunnedEffectsCleanUpEffect();
+	//Template.AddTargetEffect(CleanUpEffect);
+	DamageEffect = new class'X2Effect_ApplyWeaponDamage';
+	DamageEffect.bIgnoreBaseDamage = true;
+	DamageEffect.DamageTag = 'Volt';
+	DamageEffect.bIgnoreArmor = true;
+	DamageEffect.TargetConditions.AddItem(UnitPropertyCondition);
+	Template.AddTargetEffect(DamageEffect);
+	Template.AddMultiTargetEffect(DamageEffect);
+
+
+	Template.AssociatedPassives.AddItem('Electroshock');
+	Template.AddMultiTargetEffect(ElectroshockDisorientEffect());
+	
+	ToHitCalc = new class'X2AbilityToHitCalc_StandardAim';
+	ToHitCalc.bOnlyMultiHitWithSuccess = false;
+	ToHitCalc.BuiltInHitMod = default.CHAIN_LIGHTNING_AIM_MOD;
+	Template.AbilityToHitCalc = ToHitCalc;
+	Template.AdditionalAbilities.AddItem('CLFocus');
+	
+	Template.ActionFireClass = class'X2Action_Fire_ChainLightning';
+	Template.BuildNewGameStateFn = TypicalAbility_BuildGameState;
+	//Template.BuildVisualizationFn = class'X2Ability_SharpshooterAbilitySet'.static.Faceoff_BuildVisualization;
+	Template.BuildVisualizationFn = TypicalAbility_BuildVisualization;
+	Template.BuildInterruptGameStateFn = TypicalAbility_BuildInterruptGameState;
+	
+	Template.ChosenActivationIncreasePerUse = class'X2AbilityTemplateManager'.default.StandardShotChosenActivationIncreasePerUse;
+	Template.LostSpawnIncreasePerUse = class'X2AbilityTemplateManager'.default.StandardShotLostSpawnIncreasePerUse;
+
+	return Template;
+}
+
+
+static function X2AbilityTemplate CreateCLFocus(name AbilityName, int FocusLevel)
+{
+	local X2AbilityTemplate Template;
+	local X2Effect_ModifyTemplarFocus FocusEffect;
+
+	`CREATE_X2ABILITY_TEMPLATE(Template, AbilityName);
+
+	Template.IconImage = "img:///UILibrary_XPACK_Common.PerkIcons.UIPerk_templarFocus";
+	Template.AbilitySourceName = 'eAbilitySource_Perk';
+	Template.eAbilityIconBehaviorHUD = eAbilityIconBehavior_NeverShow;
+	Template.Hostility = eHostility_Neutral;
+	Template.bIsPassive = true;
+
+	Template.AbilityToHitCalc = default.DeadEye;
+	Template.AbilityTargetStyle = default.SelfTarget;
+	Template.AbilityTriggers.AddItem(default.UnitPostBeginPlayTrigger);
+	Template.AdditionalAbilities.AddItem('SupremeFocus');
+	Template.AdditionalAbilities.AddItem('DeepFocus');
+	Template.AdditionalAbilities.AddItem('AddSuperGigaOmegaFocus');
```

Comment:
I don't understand. Why not just have a single ability that provides the necessary bonus focus? And why make that ability require SupremeFocus?

In fact, since `CHAIN_LIGHTNING_TARGETS` is 5, doesn't that mean the Assault is getting 2 (base) + 1 + 1 + 5 focus?

Reply:
The problem is that even if focus is set to 5, it still gets clamped by default max focus being 2, so you need 3 more abilities that increase the total max focus to 5.